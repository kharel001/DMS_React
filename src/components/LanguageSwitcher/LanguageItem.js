import React from 'react';


const LanguageItem = ({language, switchLanguage, handleRequestClose}) => {
  const {icon, name} = language;
  return (
    <li className="pointer" onClick={() => {
      handleRequestClose();
      switchLanguage(language);
    }} data-test="liClick">
      <div className="d-flex align-items-center">
        <i className={`flag flag-24 flag-${icon}`}/>
        <h4 className="mb-0 ml-2" data-test="languageName">{name}</h4>
      </div>
    </li>
  );
};

export default LanguageItem;
