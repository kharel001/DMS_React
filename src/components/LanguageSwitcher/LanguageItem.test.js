import React from 'react';
import LanguageItem from './LanguageItem';
import {
    configure,
    shallow,
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})
const switchLangFunction=jest.fn();
const requestCloseFunction = jest.fn();
const dummy_data={
language:{
    icon:"icon1",
    name:"english"
},
switchLanguage:switchLangFunction,
handleRequestClose:requestCloseFunction
}

const setup = (props = {}) => {
    return shallow( < LanguageItem
     {...props}  />
    )
}

const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-test='${val}']`)
}
const findByTestAttr2=(wrapper, val)=>{
    return wrapper.find(`[data-testsss='${val}']`)
}




{/* it("Custom Scrollable render", () => {
        let wrapper = setup(dummy_data);
        // expect(wrapper).toMatchSnapshot();
}) */}
it("Snapshot LanguageItem", () => {
    let wrapper = setup(dummy_data);
    expect(wrapper).toMatchSnapshot();
    // const span2= wrapper2.find('div')
    // expect(span2).toHaveLength(5); //doesn't work as expected
    // expect(img1.props("src")).toEqual("https://via.placeholder.com/150x150")
    
})

it("Props LanguageItem", () => {
    let wrapper = setup(dummy_data);
    const languageName= wrapper.find("h4");
    expect(languageName.text()).toEqual("english")
    // const span2= wrapper2.find('div')
    // expect(span2).toHaveLength(5); //doesn't work as expected
    // expect(img1.props("src")).toEqual("https://via.placeholder.com/150x150")
    
})

it("Event Simulations LanguageItem", () => {
    let wrapper = setup(dummy_data);
    const liClick= findByTestAttr(wrapper,"liClick");
    liClick.props().onClick();
    wrapper.update();
    expect(requestCloseFunction).toHaveBeenCalled();
    expect(switchLangFunction).toHaveBeenCalledWith({
        icon:"icon1",
        name:"english"
    });

    // const span2= wrapper2.find('div')
    // expect(span2).toHaveLength(5); //doesn't work as expected
    // expect(img1.props("src")).toEqual("https://via.placeholder.com/150x150")

})