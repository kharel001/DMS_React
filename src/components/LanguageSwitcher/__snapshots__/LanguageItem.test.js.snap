// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot LanguageItem 1`] = `ShallowWrapper {}`;
